src/main/scala/nl.knaw.dans.easy.solr4files/components/Solr.scala

```diff
@@ -54,22 +59,61 @@ trait Solr extends DebugEnhancedLogging {
     }
   }
 
-  def deleteDocuments(query: String): Try[Unit] = {
-    Try(solrClient.deleteByQuery(new SolrQuery {set("q", query) }.getQuery))
-      .flatMap(checkUpdateResponseStatus)
+  def deleteDocuments(query: String): Try[UpdateResponse] = {
+    val q = new SolrQuery {
+      set("q", query)
+    }
+    Try(solrClient.deleteByQuery(q.getQuery))
+      .flatMap(checkResponseStatus)
       .recoverWith {
-        case t: HttpSolrClient.RemoteSolrException if t.getRootThrowable.endsWith("ParseException") =>
+        case t: HttpSolrClient.RemoteSolrException if isPareException(t) =>
           Failure(SolrBadRequestException(t.getMessage, t))
-        case t: SolrServerException =>
-          Failure(SolrDeleteException(query, t))
         case t =>
           Failure(SolrDeleteException(query, t))
       }
   }
 
-  def commit(): Try[Unit] = {
+  private def toJson(solrDocumentList: SolrDocumentList, query: SolrQuery): Try[String] = {
+    def toScalaMap(fieldValueMap: util.Map[String, AnyRef]) = {
```

Comment:
why not make this a proper extension method? Especially if you want to mimic `asScala` I think that is preferable.

Reply:
Will work that out when implementing data-available, I might get second thoughts about mapping versus TypeHints.